### PR TITLE
add max_length to wordlist generation

### PR DESCRIPTION
**Pull request by h00die**
This adds the ability to set the max length of a word going into a word list when written to file.

Some password hash formats have a max length (DES = 8) and therefore when generating a wordlist with anything more than that, those items will either always fail or be ignored by JTR (not sure JTR's behavior).

This currently only changes how the libraries work, i'm going through the jtr modules now and cleaning them up, writing docs, fixing logic, and implementing this behavior in other PRs.

Default behavior is `0`, which means ignore so the current behavior will stick.

With the default behavior on my system:
```
msf5 auxiliary(analyze/jtr_linux) > wc -L /tmp/jtrtmp20190114-18029-1hpr9zf
[*] exec: wc -L /tmp/jtrtmp20190114-18029-1hpr9zf

296 /tmp/jtrtmp20190114-18029-1hpr9zf
```
After setting it to 8:
```
msf5 auxiliary(analyze/jtr_linux) > wc -L /tmp/jtrtmp20190114-17944-wzzpsf
[*] exec: wc -L /tmp/jtrtmp20190114-17944-wzzpsf

8 /tmp/jtrtmp20190114-17944-wzzpsf
```

### CODE REVIEW

File: lib/metasploit/framework/jtr/wordlist.rb [hunk 1]
```diff
@@ -369,12 +369,13 @@ def mutation_keys
...
         #
+        # @param max_len [Integer] max length of a word in the wordlist, 0 default for ignored value
         # @return [Rex::Quickfile] The {Rex::Quickfile} object that the wordlist has been written to
-        def to_file
+        def to_file(max_len)
           valid!
           wordlist_file = Rex::Quickfile.new("jtrtmp")
           each_word do |word|
-            wordlist_file.puts word
+            wordlist_file.puts max_len == 0 ? word : word[0..max_len-1]
```
Comment (bcoles) [inline lib/metasploit/framework/jtr/wordlist.rb]:
```
2.3.0 :001 > word = 'asdfghjkl'
 => "asdfghjkl" 
2.3.0 :002 > max_len = 8
 => 8 
2.3.0 :003 > word[0..max_len-1]
 => "asdfghjk" 
2.3.0 :004 > word[0...max_len]
 => "asdfghjk" 
2.3.0 :005 > 
```

;)

Comment (h00die) [reply lib/metasploit/framework/jtr/wordlist.rb]: you never fail to teach me new ruby EVERY PR I submit. kudos! +1

**Comment by bcoles** [comment]
Works as described.

### Before

```
root@kali:/pentest/exploit/metasploit-framework# head /tmp/jtrtmp20190119-11675-1c89fa7
1qwerty
1qwerty
QWE123qw
QWE123qw
Q2w
Q2w
E4r
E4r
Q2w3e4r
Q2w3e4r
```

**Review by bcoles — APPROVED** [review]

**Comment by bcoles** [comment]
# Release Notes

Adds an option to set the maximum word length during JtR wordlist generation.

**Pull request merged**